airflow/kubernetes/pod_generator.py

```diff
@@ -62,76 +71,116 @@ class PodGenerator:
     :param configmaps: Any configmap refs to envfrom.
-        If more than one configmap is required, provide a comma separated list
-        configmap_a,configmap_b
-    :type configmaps: str
+    :type configmaps: Optional[List[str]]
     :param dnspolicy: Specify a dnspolicy for the pod
-    :type dnspolicy: str
-    :param pod: The fully specified pod.
-    :type pod: kubernetes.client.models.V1Pod
+    :type dnspolicy: Optional[str]
+    :param pod: The fully specified pod. Mutually exclusive with `path_or_string`
+    :type pod: Optional[kubernetes.client.models.V1Pod]
+    :param pod_template_file_or_string: Path to JSON/YAML file. Mutually exclusive with `pod`
+    :type pod_template_file_or_string: Optional[str]
+    :param extract_xcom: Whether to bring up a container for xcom
+    :type extract_xcom: bool
     """
     def __init__(  # pylint: disable=too-many-arguments,too-many-locals
         self,
-        image,
-        name=None,
-        namespace=None,
-        volume_mounts=None,
-        envs=None,
-        cmds=None,
-        args=None,
-        labels=None,
-        node_selectors=None,
-        ports=None,
-        volumes=None,
-        image_pull_policy='IfNotPresent',
-        restart_policy='Never',
-        image_pull_secrets=None,
-        init_containers=None,
-        service_account_name=None,
-        resources=None,
-        annotations=None,
-        affinity=None,
-        hostnetwork=False,
-        tolerations=None,
-        security_context=None,
-        configmaps=None,
-        dnspolicy=None,
-        pod=None,
-        extract_xcom=False,
+        image: Optional[str] = None,
+        name: Optional[str] = None,
+        namespace: Optional[str] = None,
+        volume_mounts: Optional[List[Union[k8s.V1VolumeMount, dict]]] = None,
+        envs: Optional[Dict[str, str]] = None,
+        cmds: Optional[List[str]] = None,
+        args: Optional[List[str]] = None,
+        labels: Optional[Dict[str, str]] = None,
+        node_selectors: Optional[Dict[str, str]] = None,
+        ports: Optional[List[Union[k8s.V1ContainerPort, dict]]] = None,
+        volumes: Optional[List[Union[k8s.V1Volume, dict]]] = None,
+        image_pull_policy: str = 'IfNotPresent',
+        restart_policy: str = 'Never',
+        image_pull_secrets: Optional[str] = None,
+        init_containers: Optional[List[k8s.V1Container]] = None,
+        service_account_name: Optional[str] = None,
+        resources: Optional[Union[k8s.V1ResourceRequirements, dict]] = None,
+        annotations: Optional[Dict[str, str]] = None,
+        affinity: Optional[dict] = None,
+        hostnetwork: bool = False,
+        tolerations: Optional[list] = None,
+        security_context: Optional[Union[k8s.V1PodSecurityContext, dict]] = None,
+        configmaps: Optional[List[str]] = None,
+        dnspolicy: Optional[str] = None,
+        pod: Optional[k8s.V1Pod] = None,
+        pod_template_file_or_string: Optional[str] = None,
```

Comment:
I have doubts whether this class should take this parameter if it is incompatible with all other parameters.

I also think it is worth adding extract_xcom support, because it is difficult to build a YAML file that contains the correct definition that causes the extract_xccom function to work properly

Reply:
The `extract_xcom` support works. You can see this in the [test case](https://github.com/apache/airflow/blob/c337a63786675e3bb33f6d759e4f8f882b3d0353/tests/integration/kubernetes/test_kubernetes_pod_operator.py#L591). What problems do you foresee having this as a parameter?